examples/loadspritefromsd/loadspritefromsd.c

```diff
@@ -0,0 +1,121 @@
+#include <stdio.h>
+#include <malloc.h>
+#include <string.h>
+#include <stdint.h>
+#include <libdragon.h>
+
+static const int MAX_SPRITES = 3;
+static char *sprites_rom[] = {
+	"rom:/mudkip.sprite",
+	"rom:/earthbound.sprite",
+	"rom:/plane.sprite",
+};
+static char *sprites_sd[] = {
+	"sd:/mudkip.sprite",
+	"sd:/earthbound.sprite",
+	"sd:/plane.sprite",
+};
+static bool use_sd = true;
+
+static int cur_sprite = -1;
+static sprite_t *sprite;
+
+/* Gets the file size - used to know how much to allocate */
+int filesize(FILE *pFile) {
+	fseek(pFile, 0, SEEK_END);
+	int lSize = ftell(pFile);
+	rewind(pFile);
+
+	return lSize;
+}
+
+/*
+Load a sprite with the given path. Can load both from SD and ROM.
+We can't use dfs_read here since we may be loading from the SD card.
+ */
+sprite_t *spritesheet_load(const char *sprite_path) {
+	FILE *fp = fopen(sprite_path, "r");
+	if (!fp) {
+		debugf("Error loading background\n");
+		return NULL;
+	}
+
+	const int size = filesize(fp);
+	debugf("Size: %d\n", size);
+
+	sprite_t *sprite = malloc(size);
+
+	setvbuf(fp, NULL, _IONBF, 0);
```

Comment:
Do you really need this?

Reply:
Nope, removed (along with the other unused code).